service/dap/server.go

```diff
@@ -0,0 +1,420 @@
+package dap
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"path/filepath"
+
+	"github.com/go-delve/delve/pkg/logflags"
+	"github.com/go-delve/delve/pkg/proc"
+	"github.com/go-delve/delve/service"
+	"github.com/go-delve/delve/service/api"
+	"github.com/go-delve/delve/service/debugger"
+	"github.com/google/go-dap"
+	"github.com/sirupsen/logrus"
+)
+
+// Server implements a DAP server that can accept a single client for a single
+// debug session. It does not support restarting.
+type Server struct {
+	// config is all the information necessary to start the debugger and server.
+	config *service.Config
+	// listener is used to serve HTTP.
+	listener net.Listener
+	// conn is accepted client connection.
+	conn net.Conn
+	// reader is used to read requests.
+	reader *bufio.Reader
+	// debugger is the debugger service.
+	debugger *debugger.Debugger
+	// log is used for structured logging.
+	log *logrus.Entry
+	// stopOnEntry is set to automatically stop program after start
+	stopOnEntry bool
+}
+
+// NewServer creates a new DAP Server.
+func NewServer(config *service.Config) *Server {
+	logger := logflags.DAPLogger()
+	logflags.WriteDAPListeningMessage(config.Listener.Addr().String())
+	return &Server{
+		config:   config,
+		listener: config.Listener,
+		log:      logger,
+	}
+}
+
+// Stop stops the DAP debugger service.
+// It kills the target process if it was launched by the debugger.
+func (s *Server) Stop() error {
+	s.listener.Close()
+	if s.debugger != nil {
+		kill := s.config.AttachPid == 0
+		return s.debugger.Detach(kill)
+	}
+	return nil
+}
+
+// Run starts a debugger and exposes it with an HTTP server. The debugger
```

Comment:
There's no HTTP in DAP, it's all JSON over sockets. Remove all mention of HTTP everywhere